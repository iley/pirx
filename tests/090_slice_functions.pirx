func TestResize() {
  printf(" -- resize --\n");
  var s = new([]int, 1);
  s[0] = 1;
  resize(&s, 2);
  s[1] = 2;
  printf("%d %d\n", s[0], s[1]);
}

func TestGetPtr() {
  printf(" -- getptr --\n");
  var s = new([]int8, 6);
  s[0] = 72i8;  // H
  s[1] = 101i8; // e
  s[2] = 108i8; // l
  s[3] = 108i8; // l
  s[4] = 111i8; // o
  s[5] = 0i8;

  var ptr: *int8 = getptr(s);
  printf("%s\n", getptr(s));
}

func TestSizeAndCap() {
  printf(" -- size and cap --\n");
  var s = new([]int, 8);
  printf("size = %d cap = %d\n", getsize(s), getcap(s));
  resize(&s, 9);
  printf("size = %d cap = %d\n", getsize(s), getcap(s));
}

func main(): int {
  TestResize();
  TestGetPtr();
  TestSizeAndCap();
  return 0;
}
