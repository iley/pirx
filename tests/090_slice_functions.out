 -- resize --
1 2
 -- getptr --
Hello
 -- size and cap --
size = 8 cap = 8
size = 9 cap = 16
